### ISSUE TITLE
Missing translation: en| - Unable to set language on Firefox

### ISSUE THREAD

**Issue by aaronraimist** [vector-im/element-web#9422]
Recently I've seen lots of people reporting that they are seeing missing strings for languages that are properly translated. I've also gotten bit by this a few times. The console error that looks relavent is 

```
Unable to set language 
Object { err: null, response: XMLHttpRequest }
rageshake.js:61:8
```

![Screen Shot 2019-04-10 at 10 34 12 AM](https://user-images.githubusercontent.com/5855073/55892591-2ff26d80-5b7c-11e9-98a5-9e70b1baf7eb.png)

I got it on /develop, other people on /app have reported the same issue.

**Comment by aaronraimist** [comment]
Possibly related: https://github.com/vector-im/riot-web/issues/4136 / https://github.com/vector-im/riot-web/pull/8710

**Comment by turt2live** [comment]
Looks like we might be eagerly deleting files which causes 404s later on. Might need a different cache strategy

**Comment by lampholder** [comment]
@turt2live - eagerly deleting when? When we redeploy?

**Comment by turt2live** [comment]
yes, it seems like we're blowing away the old build when we deploy a new build, which means that people who are unlucky will have missed the bundle.

**Comment by bwindels** [comment]
We could solve this by adding a service worker that adds the language files (and others) to the cache and falls back to it on a 404

**Comment by kixiQu** [comment]
This is happening for me and other people using the Riot that I host since I upgraded. The thing is, reload doesn't help. The error message is the same. Is there a header I can set with nginx to kill the caching?

**Comment by Yoric** [comment]
I keep being bitten by it several times per week, so I'd be happy if we could fix this :+1: 

Relevant console message seems to have changed:
```log
Using WebAssembly Olm
Unable to set language null KeyboardShortcuts.tsx:272:21
    n KeyboardShortcuts.tsx:272
    Z init.tsx:114
Skin loaded
```

Could we possibly init an auto-reload when we detect the issue?

**Comment by Gustry** [comment]
I got this issue many times per week as well, a popup telling me "missing translation".

But since yesterday, it **all** strings provided by Element which are pre-fixed by "Missing translation" as you can see on the screenshot

![Selection_002](https://user-images.githubusercontent.com/1609292/105040530-3eb0c880-5a62-11eb-872a-2f8ac7ac6355.jpg)

**Comment by t3chguy** [comment]
Clear your browser cache.

**Comment by Gustry** [comment]
Thanks, it worked.

**Comment by ara4n** [comment]
I'm seeing reports of this every few days atm, including from VIPs.  Can Element not load the resource via JS with a retry schedule if it fails?

**Comment by Yoric** [comment]
> Clear your browser cache.

Unfortunately, most end users know nothing about the cache, so we need a better solution.

**Comment by Gustry** [comment]
> > Clear your browser cache.
> 
> Unfortunately, most end users know nothing about the cache, so we need a better solution.

Yes, and this issue came back a few times already since my last answer. I thought it would be ok just one time, but no.

**Comment by babolivier** [comment]
I'm getting this almost every time I load Element Web now. For what it's worth, it goes away with a simple refresh of the page, but it's still pretty annoying.

**Comment by jryans** [comment]
Just to add a counterpoint, I haven't seen this in many months, at least on develop. I wonder if it's something related to cache headers...?

**Comment by babolivier** [comment]
If that helps, I should add that I'm always using develop.element.io with Firefox on Linux.

**Comment by jryans** [comment]
Okay, good to know. Unfortunately, rageshakes don't help too much for this issue at the moment, as all they show is:

> E Unable to set language null

Let's at least record the actual error to the logs...

For the moment though, the next time someone does see this on develop, please capture the actual missing file from the error, as I'd like to trace down why it's been removed.

**Comment by babolivier** [comment]
![image](https://user-images.githubusercontent.com/5547783/118645592-68df8080-b7df-11eb-998c-eb8812df3ecf.png)

^ is all I can see in the console (logs start when the Element Web tab is loaded and end a few messages after the "Unable to set language" error)

**Comment by Yoric** [comment]
Likewise,

```js
Unable to set language null rageshake.js:65:27
    fnName rageshake.js:65
    loadLanguage init.tsx:122
```

**Comment by jryans** [comment]
Okay, thanks, will look at getting more info there overall then.

**Comment by jryans** [comment]
@babolivier @Yoric Okay, develop should now be logging at least the path and status code for these errors, so please capture again next time you see it.

By the way, since you're on develop, is the typical workflow that you get the new version toast -> what's new -> update?

**Comment by babolivier** [comment]
> By the way, since you're on develop, is the typical workflow that you get the new version toast -> what's new -> update?

Yep (though sometimes I dismiss it because it's on top of something and refresh later which I assume has the same effect)

**Comment by Yoric** [comment]
I'm also using Firefox on Linux and I see the issue extremely often (I'd say 100%, but I haven't checked) at startup, when the page is restored.

**Comment by jryans** [comment]
Okay, good to have that context. For now, I'll await a fresh log report including the new path info.

**Comment by babolivier** [comment]
I've observed this once more today, the console didn't show anything new so I've rageshaken.

**Comment by babolivier** [comment]
Looks like the console does show a file name now (maybe I just wasn't paying enough attention last time):

![image](https://user-images.githubusercontent.com/5547783/119323481-07a42b00-bc7f-11eb-92f3-0e79218a4547.png)

**Comment by Yoric** [comment]
```
Unable to set language Error: Failed to load i18n/languages.7742306.json, got 404
    getLangsJson languageHandler.tsx:475
    on_end index.js:294
    on_state_change index.js:244
rageshake.js:65:27
    fnName rageshake.js:65
    loadLanguage init.tsx:122
```

**Comment by Yoric** [comment]
On my latest startup, for the first time I remember, things worked correctly. It may be related to me deactivating uBlock.

**Comment by jryans** [comment]
I checked all of the develop deployments that reference `languages.7742306.json`, and they included such a file, so as far as I can tell, the file is at least always in place on the server when it should be.

I suppose it's possible that while you're loading a the main document, a deployment is in progress, so you end up with a document reference older files that no longer exist once the deploy finishes...

@babolivier Are you using any blocking extensions?

@Yoric Do you still see the CSP log message even without uBlock? I do not see any such CSP messages here in Firefox Nightly (no blocked extensions used, but I do have Enhanced Tracking Protection set to Strict).

**Comment by babolivier** [comment]
> @babolivier Are you using any blocking extensions?

I have privacy badger (so should only block external trackers) and Firefox's tracking protection.

**Comment by Yoric** [comment]
> @Yoric Do you still see the CSP log message even without uBlock? I do not see any such CSP messages here in Firefox Nightly (no blocked extensions used, but I do have Enhanced Tracking Protection set to Strict).

I still see the CSP log message without uBlock. However, I have not encountered the Missing translation problem since then.

**Comment by Yoric** [comment]
I have the Missing translation problem today, without uBlock.
CSP log message is there.
No "Unable to set language" error on the console.

**Comment by jryans** [comment]
> I have the Missing translation problem today, without uBlock.
> CSP log message is there.
> No "Unable to set language" error on the console.

Hmm, now that's surprising that you see the missing translation problem, but yet no error in the console... 😖

**Comment by babolivier** [comment]
> > @babolivier Are you using any blocking extensions?
> 
> I have privacy badger (so should only block external trackers) and Firefox's tracking protection.

I've poked around with all tracking protections (Firefox's tracking protection and Privacy Badger) disabled for develop.element.io and I can still reproduce the error.

![image](https://user-images.githubusercontent.com/5547783/120831057-6f9c2080-c55f-11eb-9ac7-b208911f6664.png)

**Comment by Yoric** [comment]
Encountered it today again, with Firefox stable,
```
 Unable to set language Error: Failed to load i18n/languages.93b6736.json, got 404
    getLangsJson languageHandler.tsx:477
    on_end index.js:294
    on_state_change index.js:244
rageshake.js:65:27
    fnName rageshake.js:65
    loadLanguage init.tsx:122
```

**Comment by Yoric** [comment]
I realize it's a hack, but would it be possible to detect the case in which we have missing translations and reload the page after a few seconds while it's blurred?

**Comment by Kekun** [comment]
On the GNOME instance I'm seeing this since ~1h, wich is a pretty intrusive way of notifying there is a localization issue:

![Capture d’écran du 2021-06-08 13-06-07](https://user-images.githubusercontent.com/2536339/121174948-cc971f80-c85a-11eb-98c2-166ebab7fa0a.png)
![Capture d’écran du 2021-06-08 13-02-14](https://user-images.githubusercontent.com/2536339/121174951-cd2fb600-c85a-11eb-92e4-8ea6a8aced11.png)
![Capture d’écran du 2021-06-08 13-07-18](https://user-images.githubusercontent.com/2536339/121174954-cd2fb600-c85a-11eb-9e74-0c80e9c79797.png)

**Comment by jryans** [comment]
My (possibly wrong) assumption so far is that one way or another, people have ended up with an older copy of the app, which points to old content hashed JS, CSS, and languages files. The main app JS still loaded because we've configured our CDN to keep those around, but we haven't done the same for language files.

We most likely need to either:

* Tune CDN settings to actually make the language files long-lived
* Revert the language content hashing

**Comment by jryans** [comment]
> I realize it's a hack, but would it be possible to detect the case in which we have missing translations and reload the page after a few seconds while it's blurred?

My concern with this there's no way to be sure we won't be back in the same case after reload, so you either get a reload loop, or need to set some state to avoid more than one reload...

I'd much rather understand what is actually happening here.

**Comment by jryans** [comment]
After a bit of investigation, I found that our CDN treats all `.json` files (which includes the translations) as dynamic (uncacheable) content, so I believe that's part of the problem here.

We'll work on updating the cache settings, and then we can retest here.

**Comment by jryans** [comment]
I just applied some tuned cache settings for `develop.element.io`. If you use develop, please watch out for these issues over the next days / weeks, and let me know if there's any change (positive or negative) here.

**Comment by chagai95** [comment]
https://user-images.githubusercontent.com/31655082/123954885-395c9000-d9a9-11eb-960d-2dabbfafcd99.mp4

**Comment by jryans** [comment]
Hmm, very curious @chagai95, I don't think it's a persistent issue for others here (it should go away after a refresh), so that's quite surprising, and I wonder if perhaps something else is happening.

@chagai95, do you see any messages about `Unable to set language` or similar in the console?

**Comment by chagai95** [comment]
It works now so I'll check next time it happens, if...

**Comment by babolivier** [comment]
Just got this error (for the first time in a few weeks) on develop

![image](https://user-images.githubusercontent.com/5547783/125128631-8e617a00-e0fe-11eb-8a32-a1528988ee76.png)

**Comment by jryans** [comment]
Hmm, okay. 😕 At this point, I am back to not really following quite what is happening here, so I think someone else should take a look instead.

**Comment by novocaine** [comment]
> Just got this error (for the first time in a few weeks) on develop
> 
> ![image](https://user-images.githubusercontent.com/5547783/125128631-8e617a00-e0fe-11eb-8a32-a1528988ee76.png)

👋 Is this electron or web? 

If Electron, how old was the build?
If web, did it happen on load of the page, or after you'd had the tab open for a while? Did it persist after you refreshed?

**Comment by babolivier** [comment]
In web, on load, and refreshing made it go away (I don't think I've had it happen any other way before)

**Comment by unbeatable-101** [comment]
This is happening to me right now, but only with some strings and no errors in the console

**Comment by unbeatable-101** [comment]
Now I'm getting this after doing a fresh install of v1.7.34
<img width="1144" alt="Screen Shot 2021-08-04 at 02 38 41" src="https://user-images.githubusercontent.com/58151048/128133882-fb81effc-2bd3-4cb2-af62-c6cbb2e5d0c9.png">

**Comment by bjherbison** [comment]
I see an example of a missing translation on almost every new Element release on chat.mozilla.org , including the recent upgrade to v1.7.34. I might sometimes have the page open when an upgrade happens, but usually (and the most recent time) it was a freshly opened web page after the update.
I used Firefox, both the release version and the Nightly version. I haven't paid attention to which versions of Firefox I see the problem with. (I've just trained myself to use control--shift-R whenever I think there has been an update).
I will try to remember this issue and try to gather more information the next time I see the issue.

**Comment by t3chguy** [comment]
I believe the issue with EMS deployments is that they don't keep around the older version files, so if your browser already has some files and tries to load the rest they won't be there anymore.

**Comment by bjherbison** [comment]
> I believe the issue with EMS deployments is that they don't keep around the older version files, so if your browser already has some files and tries to load the rest they won't be there anymore.

Good. That suggests three easy fixes: Either change the upgrade procedure so the older files are kept, change the web site code to notice a new release and force an update of the necessary files/information, or add version numbers to file names as needed so the correct versions are always used.

**Comment by t3chguy** [comment]
> or add version numbers to file names as needed so the correct versions are always used.

This is already done, but if your browser has Element v1.7.33 loaded, and v1.7.34 has been deployed then it'll be looking for English.1.7.33.json (we actually use file hashes, not versions, but the effect is the same) which no longer exists.

**Comment by bjherbison** [comment]
Seriously, there's no way for the browser software to figure out the deployed version is newer and tell the user an update is needed? If nothing else the missing 1.7.33.json should be a clue.

**Comment by t3chguy** [comment]
Yup, would require quite a large bunch of work making use of Service Workers. Otherwise the browser cache will give us a split-brained view of the world. It is definitely planned but quite a large commitment. We have tried it once in the past and broke update prompts altogether so had to back it out.

**Comment by novocaine** [comment]
> Seriously, there's no way for the browser software to figure out the deployed version is newer and tell the user an update is needed? If nothing else the missing 1.7.33.json should be a clue.

I agree. I think at the very least we should be able to flag the situation to the user to take action rather than just dying.

**Comment by novocaine** [comment]
I'm investigating whether this could be solved by tweaking some caching headers on the root html page.

**Comment by novocaine** [comment]
> Seriously, there's no way for the browser software to figure out the deployed version is newer and tell the user an update is needed? If nothing else the missing 1.7.33.json should be a clue.

We now return `cache-control: no-cache` from `/` for `{dev, app, staging, develop}.element.io` which causes the browser to explicitly check for a new version of the html page, which contains a link to the app bundle named by its hash. Previously, without this header, on page load, browsers would simply load the page from disk without checking for a new version - so updates to new versions of the app were missed for long periods. 

This doesn't entirely solve the underlying issue. It'll still be possible to fail to dynamically load an asset (like a new translation dictionary) if you're on an old version. But I am hoping this greatly reduces the likelihood of it occurring for the manner it's being its encountered by reporters here, which is on page load.

EMS is working on a change for their deployments.

I am not sure whether chat.mozilla.org is hosted by Mozilla or by EMS - if it's the former, Mozilla may need to update their own config to see the same benefit.

I will leave this issue open to monitor for new reports post this change.

**Comment by kittykat** [comment]
@novocaine got the issue a couple of days ago on a netlify build and local build. Both times on web. In both cases page/dialog were showing fine, and then at some point on page refresh started showing the missing translations which were broken on page/dialog load and refresh didn't fix it.

**Comment by novocaine** [comment]
Local and Netlify builds have their own complications in terms of assets becoming out of sync. For local, its likely that i18n assets need to be rebuilt; for Netlify I'm not sure how the translations are uploaded, but it'll be quite different to prod issues and we can track the issue separately as a developer experience ticket as end users don't use those

**Comment by ara4n** [comment]
folks are still getting this...

![image](https://user-images.githubusercontent.com/1294269/142915030-ae54e07c-470c-4870-a9a8-2c6394701044.png)

**Comment by novocaine** [comment]
```
2021-11-22T18:01:10.160Z E Unable to set language Failed to load i18n/languages.9442f49.json, got 404
O/</<@https://app.element.io/bundles/d443354bdb439e133de8/vendors~init.js:2:90104
t/</i.onreadystatechange/<@https://app.element.io/bundles/d443354bdb439e133de8/vendors~init.js:2:28999
t/</i.onreadystatechange@https://app.element.io/bundles/d443354bdb439e133de8/vendors~init.js:2:29023
```

**Comment by ShadowJonathan** [comment]
> when restarting Firefox with an element.io tab in it

For the record, I've gotten this same issue around crashing and restarting my chromium browser, but it always seemed to happen to develop, so I chalked it up to non-prod cache settings, a refresh seemed to always fix it.

**Comment by novocaine** [comment]
> > when restarting Firefox with an element.io tab in it
> 
> For the record, I've gotten this same issue around crashing and restarting my chromium browser, but it always seemed to happen to develop, so I chalked it up to non-prod cache settings, a refresh seemed to always fix it.

If this is still happening since Sept 10, it shouldn't be, even on develop, so a rageshake would be useful next time it occurs

**Comment by babolivier** [comment]
Caught one in the wild today, have sent a rageshake

**Comment by ShadowJonathan** [comment]
Just got it as well, sent rageshake

**Comment by novocaine** [comment]
Cache invalidation is not occurring if people reload Element from a url containing a question mark, and so may encounter this issue in that case. See https://github.com/vector-im/element-web/issues/20033#issuecomment-986656638

**Comment by novocaine** [comment]
> Cache invalidation is not occurring if people reload Element from a url containing a question mark, and so may encounter this issue in that case. See [#20033 (comment)](https://github.com/vector-im/element-web/issues/20033#issuecomment-986656638)

hopefully resolved by https://github.com/vector-im/element-web/issues/20033#issuecomment-1012990617

**Comment by lindhe** [comment]
Got hit by this today. :/ Are we getting any closer to a solution?

**Comment by novocaine** [comment]
> Got hit by this today. :/ Are we getting any closer to a solution?

Thanks for the report. The issue is multi-faceted so we need your help to understand what caused it in your particular case.

* What browser are you using?
* What were your exact actions prior to encountering the issue? How did you load the page?
* Did the problem resolve itself after you reloaded the browser?

**Comment by Yoric** [comment]
I've stopped reporting this every time I hit this, should I?

**Comment by novocaine** [comment]
> I've stopped reporting this every time I hit this, should I?

You should report it here with the details above and a rageshake please.

**Comment by novocaine** [comment]
> > Thanks for the report. The issue is multi-faceted so we need your help to understand what caused it in your particular case.
> > ```
> > * What browser are you using and what homeserver?
> > 
> > * What were your exact actions prior to encountering the issue? How did you load the page, and from what exact URL did you load it?
> > 
> > * Did the problem resolve itself after you reloaded the browser?
> > ```
> 
> I'm using latest stable Firefox (currently on 96.0), but I know that I've had this issue a dozen times or so the past year.
> 
> I'm on the matrix.org homeserver.
> 
> I'm not quite sure what I did before encountering it. I have a feeling it's most frequent when there's an update prompting me to reload the page/app in the browser, and/or possibly when I open firefox with the element messenger open in a tab (i.e. possibly stale cache).
> 
> Yes, it has always resolved after reloading the page.

Thankyou! Are you using app.element.io ?

**Comment by lindhe** [comment]
Yes.

**Comment by Yoric** [comment]
Got it yesterday and today, see https://github.com/matrix-org/element-web-rageshakes/issues/9880 and https://github.com/matrix-org/element-web-rageshakes/issues/9985 .

> - What browser are you using and what homeserver?

Firefox 96.0 (64-bit), Ubuntu version. My account is on matrix.org.

> - What were your exact actions prior to encountering the issue? How did you load the page, and from what exact URL did you load it?

Starting the browser. It always happens to me when I restore a session. On the other hand, I almost never open element-web manually (except when testing new features) because it opens with my browser.

> - Did the problem resolve itself after you reloaded the browser?

Yup.

**Comment by lindhe** [comment]
I got it again right now. Started up Firefox 96 with `https://app.element.io/#/room/!nazbIMvikFakSzBrGP:matrix.org` (e2ee 1-on-1 chat room) as a pinned tab. It just hangs on the loading screen. Even refreshing with F5 doesn't help, I have to go https://app.element.io to get it right again.
| Web page | Developer Console |
| :-: | :-: |
| ![Screenshot 2022-02-05 at 16-54-25 Element](https://user-images.githubusercontent.com/7773090/152649016-9e4cc339-1edc-41dd-856f-2eaa42c15d78.png) | ![Screenshot from 2022-02-05 16-55-00](https://user-images.githubusercontent.com/7773090/152649147-7f8572bc-7046-4ef5-92d4-1fd4ae6256ce.png) |

**Comment by babolivier** [comment]
Got the error again (same versions as last time), I rageshook: https://github.com/matrix-org/element-web-rageshakes/issues/10498

Also as @Yoric points out it also seems to be only happening when I restore a session when opening up Firefox.

**Comment by andybalaam** [comment]
I got this again today and rageshook. (I get it pretty much every day.)

Firefox 96.0 on Ubuntu 21.10, with https://develop.element.io/#/room/!ePQtFybKgQDLPPafFg:matrix.org as a pinned tab.
Element version: ba55473a0e4e-react-c19aa957b655-js-74d24f38f742
Olm version: 3.2.8

It resolved with a page refresh.

**Comment by lindhe** [comment]
Had it again. As always for me, it was just after opening Firefox. Not reproducable for me by just restarting Firefox, possibly I have to restart my computer to make the cache go away. Or a timeout, I don't know.

**Comment by andybalaam** [comment]
Got this again today on https://develop.element.io/#/room/!babyLsOltZZoqvPNrK:one.ems.host .
I rageshook.

**Comment by novocaine** [comment]
This is being reported fairly regularly on Firefox so bumping the frequency.

At this point, if someone gets this *not* on Firefox, we might want a different issue.

**Comment by andybalaam** [comment]
Happened to me again this morning, rageshake ^

**Comment by andybalaam** [comment]
And again today, after restarting my computer, on Firefox 97.0.2. Rageshake ^

**Comment by babolivier** [comment]
Since it sounds like we have identified the source of the bug (ie the underlying [Firefox bug](https://bugzilla.mozilla.org/show_bug.cgi?id=706970), are rageshakes still necessary?
Also, what's the timeline for implementing a workaround? It's frustrating that the cause isn't Element but it still makes the UX of using Element on Firefox pretty bad.

**Comment by novocaine** [comment]
https://github.com/vector-im/element-web/pull/21272 was an attempt to fix it. It would be helpful to understand whether this actually kicked in and whether the page reloaded with ?updated=1 or not @andybalaam

**Comment by andybalaam** [comment]
Hmm.  I didn't notice, and have accepted an update since then. I will clear my cache and report back if this happens again.

**Comment by babolivier** [comment]
Ah, I wasn't aware that a mitigation had been merged in. I've definitely seen the bug happening again this morning on two Element tabs but didn't rageshake because I thought it wasn't necessary, I'll do it next time I reproduce it.

**Comment by babolivier** [comment]
I've seen it again just now, have rageshaken ^

**Comment by andybalaam** [comment]
I see it this morning, rageshake above ^.

The location bar in my browser does not include `?updated=1`

**Comment by andybalaam** [comment]
Seeing it again today, no `?updated=1` in the location bar

**Comment by Yoric** [comment]
> Seeing it again today, no `?updated=1` in the location bar

Likewise. Should I continue rageshakes?

**Comment by novocaine** [comment]
There's another PR up at https://github.com/vector-im/element-web/pull/21410, please wait for that one

**Comment by babolivier** [comment]
I've seen it happen again yesterday and today, have rageshaken ^

**Comment by novocaine** [comment]
the rageshake is really helpful, thanks. it looks like the out of date app is being detected correctly, but the forced reload isn't happening still.

relevant lines

> 2022-03-24T21:03:23.873Z I startUpdater, current version is e638abf7c253-react-3bb0dc08e81e-js-6192325fe0b6
> 2022-03-24T21:03:23.924Z I Update available to e638abf7c253-react-f229ad640714-js-c541b3f1ce0f, will notify user
> 2022-03-24T21:03:23.924Z I reloadPage to [redacted] <-- reload should happen here, but appears to not

I will try to unpick that when I get some time next week.

**Comment by babolivier** [comment]
In case it helps, it's happened to me again today, have rageshaken ^

**Comment by cpeterso** [comment]
Suggestion:

When translations are missing and Element shows the English fallback strings, just show the fallback strings without the `missing translation: en|` prefix (unless some Element developer setting is set). Example: show `New version of Element is available` instead of `missing translation: en|New version of Element is available`.

Or maybe just omit `missing translation: en|` for users already using an English locale.

`missing translation: en|` adds no value to users: there is nothing they can do about the error and it makes the client unusable.

**Comment by babolivier** [comment]
I've seen it happen on develop just now, have rageshaken ^

**Comment by andybalaam** [comment]
I got this again this morning, rageshake ^

**Comment by andybalaam** [comment]
Got this again today.

**Comment by andybalaam** [comment]
Just seen it again today, with `updated=1` in the URL :-(

**Comment by novocaine** [comment]
> Just seen it again today, with `updated=1` in the URL :-(

rageshake please

**Comment by andybalaam** [comment]
> rageshake please

Sorry, forgot to mention that rageshake immediately above that comment is mine.

**Comment by t3chguy** [comment]
> Yeah, this is still present. I got it and pressed "send feedback". Hope that's equivalent to rage shaking. I tried rage shaking my laptop, but nothing happened...

Unfortunately feedback doesn't include logs for privacy reasons, please send logs from Settings > Help & About

**Comment by novocaine** [comment]
> Yeah, this is still present. I got it and pressed "send feedback". Hope that's equivalent to rage shaking. I tried rage shaking my laptop, but nothing happened...

Element `1.10.11` was created on April 19, so doesn't have the [most recent attempt to fix the issue](https://github.com/vector-im/element-web/pull/21912)

**Comment by babolivier** [comment]
I saw it again this morning on develop, I've rageshaken ^

**Comment by babolivier** [comment]
It's happened to me again this morning, except this time it seemed to prevent Element to load (I got an infinite spinner until I reloaded the tab), so I couldn't rageshake.

**Comment by andybalaam** [comment]
I got the infinite spinner this morning. Console:

```
Content Security Policy: The page's settings blocked the loading of a resource at inline ("script-src"). moz-extension:33:9
Unable to set language Error: Failed to load i18n/languages.630445c.json, got 404
    getLangsJson languageHandler.tsx:533
    on_end index.js:294
    on_state_change index.js:244
rageshake.ts:72:27
Loading Worker from “https://develop.element.io/ad35579c7db15f5fe93e.worker.js” was blocked because of a disallowed MIME type (“text/html”). develop.element.io
```

Address bar:

```
https://develop.element.io/?updated=850a250cea91-react-1c70696b1020-js-d190cdc307f1#/room/!LjeDfEINUDXfGPPsHm:matrix.org
```

The screen is blank except for "missing translation: en|Logout" at the bottom, and a spinner in the middle.

**Comment by babolivier** [comment]
I got it again this morning. I managed to rageshake through the console ^

**Comment by andybalaam** [comment]
I got it again today (not the freeze), rageshake ^, URL: https://develop.element.io/?updated=850a250cea91-react-1c70696b1020-js-d190cdc307f1#/room/!HtSAtjJmFcbeysRBii:envs.net

**Comment by babolivier** [comment]
Also had one just now (not the freeze), rageshaken ^

**Comment by babolivier** [comment]
Again ^

**Comment by turt2live** [comment]
It's unclear what this was blocked on, but it's certainly worthy of picking up. 

I believe at this point we know what causes it, we just need to figure out how to fix it.

**Comment by andybalaam** [comment]
I haven't seen it for ages. Closing. If it re-occurs please re-open. (Thanks Rich for pointing this out.)
